### PR TITLE
v2raya: v2rayA should start after nftables

### PR DESCRIPTION
**Pull request by Eliot00**
###### Description of changes

https://github.com/v2rayA/v2rayA/pull/723

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Eliot00** [comment]
https://github.com/NixOS/nixpkgs/pull/204807#discussion_r1045273722

**Comment by Eliot00** [comment]
https://github.com/NixOS/nixpkgs/pull/204807#discussion_r1045273722

**Comment by Eliot00** [comment]
https://github.com/NixOS/nixpkgs/pull/204807#discussion_r1045273883

File: nixos/modules/services/networking/v2raya.nix [hunk 1]
```diff
@@ -16,7 +16,12 @@ with lib;
       unitConfig = {
         Description = "v2rayA service";
         Documentation = "https://github.com/v2rayA/v2rayA/wiki";
-        After = [ "network.target" "nss-lookup.target" "iptables.service" "ip6tables.service" ];
+        After = [
+          "network.target"
+          "nss-lookup.target"
+          "iptables.service"
+          "nftables.service"
```
Comment (Vonfry) [inline nixos/modules/services/networking/v2raya.nix:23]: Why dose v2raya start after both nftables and iptables? Usually, only one of them is used, and v2raya will never be started if both of them is in after. It is better to use `config.networking.nftables.enable` or something like it to determine which one is used.

File: nixos/modules/services/networking/v2raya.nix [hunk 2]
```diff
@@ -12,27 +12,38 @@ with lib;
...
+          LimitNOFILE = 1000000;
+          Restart = "on-failure";
+          Type = "simple";
+        };
 
-      wantedBy = [ "multi-user.target" ];
-      path = with pkgs; [ iptables bash iproute2 ]; # required by v2rayA TProxy functionality
-    };
+        wantedBy = [ "multi-user.target" ];
+        path = with pkgs; [ iptables bash iproute2 ]; # required by v2rayA TProxy functionality
```
Comment (Vonfry) [inline nixos/modules/services/networking/v2raya.nix:45]: Does TProxy only work with iptables? Can it work with nftables? If a user choose nftables and he open tproxy, can tproxy work as expect?

**Review by Vonfry — COMMENTED** [review]
IMO, the module changes and package changes should be split into two different commit.

And other diffs look good to me.

File: nixos/modules/services/networking/v2raya.nix (hunk 2 above)
Comment (Eliot00) [reply nixos/modules/services/networking/v2raya.nix:45]: It seems that nftables is not currently supported https://github.com/v2rayA/v2rayA/issues/293

**Comment by Vonfry** [comment]
ping @NickCao .

**Review by NickCao — APPROVED** [review]

**Pull request merged**